internal/gengapic/paging.go

```diff
@@ -73,16 +80,23 @@ func (g *generator) iterTypeOf(elemField *descriptor.FieldDescriptorProto) (*ite
 				return nil, errors.E(nil, "unusual map entry message: %q", eMsg)
 			}
 
+			// The most common case is mapping to messages,
+			// but check in case it's a primitive.
 			if valueField.GetType() == descriptor.FieldDescriptorProto_TYPE_MESSAGE {
-				pt.mapValueTypeName = valueField.GetTypeName()
+				vType := g.descInfo.Type[valueField.GetTypeName()]
+				imp, err := g.descInfo.ImportSpec(vType)
+				if err != nil {
+					return nil, err
+				}
+
+				elems := strings.Split(valueField.GetTypeName(), ".")
```

Comment:
If you use `NameSpec` it will give the Go name for the type. I think that's what you're trying to do with all of the splitting?

Reply:
Correct. I'll make another pass and use `NameSpec`, since it's taking care of all the string shenanigans for me.